/* Copyright 2020 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
import "fedlearner/common/common.proto";

package fedlearner.common;

message SyncRequest {}

message SyncResponse {
  int64 file_index = 1;
  int64 row_index = 2;
}

message Request {
  Status status = 1;
  int64 file_index = 2;
  int64 row_index = 4;
  int64 row_num = 5;
  bytes payload = 7;
}

message Response {
  Status status = 1;
  int64 file_index = 2;
  int64 row_index = 3;
  int64 row_num = 4;
  bytes payload = 5;
}

service StreamTransmitService {
  rpc SyncState(SyncRequest) returns (SyncResponse) {}
  rpc Transmit(stream Request) returns (stream Response) {}
}
